hack/tidy-check.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+# Copyright 2019 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set +e
+THIS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+PROJECT_DIR=$(dirname "$THIS_DIR")
+
+pushd "$THIS_DIR" >/dev/null || exit
+
+MOD_FILE="$PROJECT_DIR/go.mod"
+SUM_FILE="$PROJECT_DIR/go.sum"
+TMP_DIR="$THIS_DIR/.tmp.tidy-check"
+TMP_MOD_FILE="$TMP_DIR/go.mod"
+TMP_SUM_FILE="$TMP_DIR/go.sum"
+TARGET_GO_VER="1.12"
+TARGET_GO_VER_PATTERN="go$TARGET_GO_VER.*"
+DOCKER_GO="docker run --rm -w /antrea -v $PROJECT_DIR:/antrea golang:$TARGET_GO_VER go"
+
+if [ "$(uname -s)" = Linux ]; then # Allow user to specify the path to `go` on linux
```

Comment:
I think this whole `if` block should just be:
```
# if Go environment variable is set, use it as it is, otherwise default to "go"
: "${GO:=go}"
```

Reply:
Basically if `$GO` is set to the "correct" version (1.12), use that. Otherwise if `go` is the correct version, use that. Otherwise use the docker.